framework/packages/abstract-client/src/account.rs

```diff
@@ -0,0 +1,162 @@
+    }
+
+    pub fn name(self, name: impl Into<String>) -> Self {
+        Self {
+            name: Some(name.into()),
+            ..self
+        }
+    }
+
+    pub fn description(self, description: impl Into<String>) -> Self {
+        Self {
+            description: Some(description.into()),
+            ..self
+        }
+    }
+
+    pub fn link(self, link: impl Into<String>) -> Self {
+        Self {
+            link: Some(link.into()),
+            ..self
+        }
+    }
+
+    pub fn namespace(self, namespace: impl Into<String>) -> Self {
+        Self {
+            namespace: Some(namespace.into()),
+            ..self
+        }
+    }
+
+    pub fn base_asset(self, base_asset: AssetEntry) -> Self {
+        Self {
+            base_asset: Some(base_asset),
+            ..self
+        }
+    }
+
+    pub fn governance_details(self, governance_details: GovernanceDetails<String>) -> Self {
+        Self {
+            governance_details: Some(governance_details),
+            ..self
+        }
+    }
+
+    pub fn build(self) -> AbstractClientResult<Account<Chain>> {
+        let sender = self.environment().sender().to_string();
+        let name = self
+            .name
+            .unwrap_or_else(|| String::from("Default Abstract Account"));
+        let governance_details = self
+            .governance_details
+            .unwrap_or(GovernanceDetails::Monarchy { monarch: sender });
+        let abstract_account = self.abstr.account_factory.create_new_account(
+            AccountDetails {
+                name,
+                description: self.description,
+                link: self.link,
+                namespace: self.namespace,
+                base_asset: self.base_asset,
+                install_modules: vec![],
+            },
+            governance_details,
+            Some(&[]),
+        )?;
+        Ok(Account::new(abstract_account))
+    }
+}
+
+pub struct Account<Chain: CwEnv> {
+    pub(crate) abstr_account: AbstractAccount<Chain>,
+}
+
+impl<Chain: CwEnv> Account<Chain> {
+    pub(crate) fn new(abstract_account: AbstractAccount<Chain>) -> Self {
+        Self {
+            abstr_account: abstract_account,
+        }
+    }
+
+    pub(crate) fn from_namespace(
+        abstr: &Abstract<Chain>,
+        namespace: String,
+    ) -> AbstractClientResult<Self> {
+        let namespace_response: NamespaceResponse = abstr
+            .version_control
+            .namespace(Namespace::new(&namespace)?)?;
+
+        let abstract_account: AbstractAccount<Chain> =
+            AbstractAccount::new(abstr, Some(namespace_response.account_id));
+
+        Ok(Self::new(abstract_account))
+    }
+
+    pub fn get_account_info(&self) -> AbstractClientResult<AccountInfo<Addr>> {
+        let info_response: InfoResponse = self.abstr_account.manager.info()?;
+        Ok(info_response.info)
+    }
+
+    // Install an application on the account
+    // creates a new sub-account and installs the application on it.
```

Comment:
Does a sub account automatically get created when we install an app on an account?

Reply:
No, you need to create the sub-account and then install the app on that sub-account.

Reply:
Will do that then, thanks for the clarification :)